aten/src/ATen/cuda/CUDAGenerator.cpp

```diff
@@ -44,25 +45,25 @@ CUDAGenerator* getDefaultCUDAGenerator(DeviceIndex device_index) {
     TORCH_CHECK(idx >= 0 && idx < num_gpus);
   }
   std::call_once(cuda_gens_init_flag[idx], [&] {
-    default_gens_cuda[idx] = std::make_shared<CUDAGenerator>(idx);
+    default_gens_cuda[idx] = make_generator<CUDAGenerator>(idx);
     default_gens_cuda[idx]->seed();
   });
-  return default_gens_cuda[idx].get();
+  return default_gens_cuda[idx];
 }
 
 /**
  * Utility to create a CUDAGenerator. Returns a shared_ptr
  */
-std::shared_ptr<CUDAGenerator> createCUDAGenerator(DeviceIndex device_index) {
+Generator createCUDAGenerator(DeviceIndex device_index) {
   std::call_once(num_gpu_init_flag, initCUDAGenVector);
   DeviceIndex idx = device_index;
   if (idx == -1) {
     idx = c10::cuda::current_device();
   }
   TORCH_CHECK(idx >= 0 && idx < num_gpus, "The device_index is invalid.");
-  auto gen = std::make_shared<CUDAGenerator>(idx);
-  gen->set_current_seed(default_rng_seed_val);
-  gen->set_philox_offset_per_thread(0);
+  auto gen = make_generator<CUDAGenerator>(idx);
+  check_generator<CUDAGenerator>(gen)->set_current_seed(default_rng_seed_val);
```

Comment:
This is kind of goofy. Better to just check generator once and then use the resulting handle to do the mutations you need, right?

Reply:
addressed